### PR TITLE
added adding files in github with auth token

### PR DESCRIPTION
**Pull request by AbhayDhaundiyal**

### CODE REVIEW

**Comment by AbhayDhaundiyal** [comment]
Closes #6

File: common/common.go [hunk 1]
```diff
@@ -77,63 +77,70 @@ func (t *Tree) TraverseTree(node *Node) {
 	if node == nil {
 		return
 	}
-	fmt.Println(node.Path)
-	CreateRefOnGitHub(node, t.GitHubRepo)
+	vv, err := node.Info.Info()
```
Comment (Regis-Caelum) [inline common/common.go]: Use appropriate variable name

File: common/common.go [hunk 2]
```diff
@@ -77,63 +77,70 @@ func (t *Tree) TraverseTree(node *Node) {
...
+	if err != nil {
+		return
+	}
+	if !vv.IsDir() {
```
Comment (Regis-Caelum) [inline common/common.go]: implement this logic under `CreateRefOnGitHub` function

File: common/common.go [hunk 3]
```diff
@@ -77,63 +77,70 @@ func (t *Tree) TraverseTree(node *Node) {
...
 func CheckErr(err error) {
 	if err != nil {
-		log.Printf("Error: %v", err.Error())
+		fmt.Printf("Error: %v", err.Error())
 	}
 }
 
-func CreateRefOnGitHub(node *Node, gitHubUrl string) {
-	// TODO: Implement function to create the reference on GitHub Repo
+func CreateRefOnGitHub(gitPath string, node *Node, gitHubUrl string) {
```
Comment (Regis-Caelum) [inline common/common.go]: Don't change the function defination, just implement it

File: main.go [hunk 4]
```diff
@@ -8,11 +8,10 @@ import (
 )
 
 func main() {
-	path := "/home/mehmed/Desktop"
-	gitHubUrl := "https://github.com/Regis-Caelum/Backup/contents"
+	path := "/home/kenkaneki124/dronebase"
```
Comment (Regis-Caelum) [inline main.go]: use env variables for sensitive info

**Review by Regis-Caelum — CHANGES_REQUESTED** [review]
Please do the changes and LMK when youre done

**Comment by Regis-Caelum** [comment]
please provide description for the PR and use significant commit messages

File: main.go (hunk 4 above)
Comment (AbhayDhaundiyal) [reply main.go]: Done!

**Comment by AbhayDhaundiyal** [comment]
I have resolved the issue please review it @Regis-Caelum

**Review by Regis-Caelum — APPROVED** [review]
Nice,
Keep up the good work :+1:

**Pull request merged**